Modules/Module_Import/Import/xASL_imp_DCM2NII_Subject_SortASLVolumes.m

```diff
@@ -146,6 +146,21 @@
         elseif  ~isempty(strfind(scan_name,'M0'))
             [nii_files,ASLContext] = xASL_bids_MergeNifti(nii_files, 'M0', niiTable);
         end
+	else
+		% In case of a single file, we need to check issues with orientation
+		% Check for a special case of Siemens MOSAIC files without Phoenix field
+		if isfield(tmpJSON, 'Manufacturer') && strcmpi(tmpJSON.Manufacturer, 'Siemens') && isfield(tmpJSON, 'ImageType') && ~isempty(regexpi(tmpJSON.ImageType, 'MOSAIC')) && (~isfield(tmpJSON, 'PhoenixProtocol') || isempty(tmpJSON.PhoenixProtocol))
+			niftiData = xASL_io_ReadNifti(nii_files{1});
+			% We check for a special case of image center in the Z-direction not being crossing 0
+			if (sum(niftiData.mat(:,3)) > 0) && (niftiData.mat(3,4) > 0)
+				% The image is mirrored in the Z-direction, we need to fix the image matrix and the orientation matrix
+				im = xASL_io_Nifti2Im(nii_files{1});
+				im = im(:,:,end:-1:1,:,:,:);
+				mat = niftiData.mat;
```

Comment:
You can move this line up so we have:

if (sum(mat(:,3)) > 0) && (mat(3,4) > 0)

Reply:
Done